plugins/CoreAdminHome/API.php

```diff
@@ -121,22 +121,33 @@ public function setTrustedHosts($trustedHosts)
     /**
      * @internal
      */
-    public function setBrandingSettings($useCustomLogo)
+    public function setBrandingSettings($useCustomLogo, $hasCustomLogo, $hasCustomFavicon)
     {
         Piwik::checkUserHasSuperUserAccess();
-
         $customLogo = new CustomLogo();
-
-        if ($customLogo->isCustomLogoFeatureEnabled()) {
-            if ($useCustomLogo) {
-                $customLogo->enable();
-            } else {
-                $customLogo->disable();
+        $response = [];
+
+        if (!$useCustomLogo || ($useCustomLogo && !$hasCustomLogo && !$hasCustomFavicon)) {
+            $customLogo->removeLogos();
+            $customLogo->disable();
+
+            $response['useCustomLogo'] = false;
```

Comment:
```suggestion
            $response['useCustomLogo'] = false;
            return $response;
   }
```
You could use an early return here, and remove the elseif as it is a bit redundant.